jalantechnologies/boilerplate-mern
# Build system updates

## Description
- [feat] Added support for building frontend (webpack + tsc) and backend (tsc) builds via `npm run build`
- [feat] Added support for hot reloading frontend and backend builds using separate processes via `npm run serve`
- [feat] Added support for serving static assets from `src/assets` at `/assets`.
- [feat] Added support for injecting `public` config in frontend builds. Application can use `window.Config` to access the same.
- [feat] Added support for injecting frontend services using React context
- [feat] Added support for `react-dom-router` in React app
- [dev] Fixed issue where eslint was not working for `tsx` and `/cypress` files
- [dev] Changed development docker image architecture to `linux/amd64` to support M1+ Macs when running compose in development
- [dev] MongoDb driver logs are now written to a log file instead of cluttering `stdout` when running compose in development
- [dev] Added support for persisting MongoDb data files when running compose in development
- [chore] Restructured frontend (`src/apps/frontend`) and backend (`src/apps/backend`) code into respective directories
- [chore] Changed npm script for running E2E test cases from `e2e` to `test:e2e`
- [chore] Deleted unused and outdated CircleCI config

## Database schema changes
- NA

## Tests
### Automated test cases added
- NA

### Manual test cases run
- Running `npm run build` and `npm start` should run the server in production environment
- Running `npm run serve` should watch frontend and backend code in receptive processes and run server in development environment
- Running `npm test` should work same as before
- Running `npm run test:e2e` should work same as `npm run e2e` as was before
- Running `docker compose -f docker-compose.dev.yml up` should work same as before
